config/config_test.go

```diff
@@ -709,38 +607,28 @@ func TestGetProxySettings(t *testing.T) {
 }
 
 func TestEnvSiteConfig(t *testing.T) {
-	ddconfig.Datadog.Set("process_config.process_dd_url", "")
+	origcfg := config.Datadog
+	config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	defer func() {
+		config.Datadog = origcfg
+	}()
+
 	assert := assert.New(t)
-	for _, tc := range []struct {
-		site     string
-		ddURL    string
-		expected string
-	}{
-		{
-			"datadoghq.io",
-			"",
-			"process.datadoghq.io",
-		},
-		{
-			"",
-			"https://process.datadoghq.eu",
-			"process.datadoghq.eu",
-		},
-		{
-			"datacathq.eu",
-			"https://burrito.com",
-			"burrito.com",
-		},
-	} {
-		// Fake the os.Setenv("DD_SITE", tc.site)
-		ddconfig.Datadog.Set("site", tc.site)
-		os.Setenv("DD_PROCESS_AGENT_URL", tc.ddURL)
 
-		agentConfig, err := NewAgentConfig(nil, &YamlAgentConfig{}, nil)
-		assert.NoError(err)
-		assert.Equal(tc.expected, agentConfig.APIEndpoints[0].Endpoint.Hostname())
-	}
+	config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	agentConfig, err := NewAgentConfig(nil, &NetworkAgentConfig{}, "./testdata/TestEnvSiteConfig.yaml")
```

Comment:
I believe this test was to test the behavior when we had `site` set in env variables too as well as the `process_dd_url` variable in the configuration, could we add this back ?

Reply:
Worth doing for a single test, but seems messy to do it X times. I'll add it back for a single case

Reply:
👍